Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs

```diff
@@ -231,13 +239,75 @@ public void DeleteCurvePoints(int[] pointsIndeces)
                     break;
             }
             _powerSHAPE.DoCommand("REPOINT_CURVE",
+                                  "TOLERANCED OFF",
                                   "START " + startPoint,
                                   "END " + endPoint,
                                   "NUMBER " + numberOfPoints,
                                   "CORNERS " + corners,
                                   "ACCEPT");
         }
 
+        /// <summary>
+        /// Repoints the curve between the specified points to the specified tolerance.
+        /// </summary>
+        /// <param name="startPoint">The index of the start point. Index for the points is zero based.</param>
+        /// <param name="endPoint">The index of the end point. Index for the points is zero based.</param>
+        /// <param name="tolerance">The maximum distance that the repointed curve may vary from the original curve.</param>
+        /// <param name="cornerBehaviour">You can select points which you don't want to remove. These points are marked as corners.</param>
+        public void RepointCurveBetweenPoints(
+            int startPoint,
+            int endPoint,
+            float tolerance,
+            CornerBehaviours cornerBehaviour = CornerBehaviours.SelectCorners)
+        {
+            AddToSelection(true);
+
+            startPoint = startPoint + 1;
+            endPoint = endPoint + 1;
+
+            if (startPoint < 1 || startPoint > NumberPoints)
+            {
+                throw new Exception("Start Point is not defined");
+            }
+            if (endPoint < 1 || endPoint > NumberPoints)
+            {
+                throw new Exception("End Point is not defined");
+            }
+            if (tolerance < 1e-5)
```

Comment:
Is this the right tolerance or should you compare it to the general tolerance in PowerShape?

Reply:
This tolerance can be specified independently of the session's general tolerance. The minimum value supported is 1e-5.